android/src/main/java/com/pilloxa/dfu/RNNordicDfuModule.java

```diff
@@ -38,7 +38,13 @@ public void startDFU(String address, String name, String filePath, Promise promi
             starter.setDeviceName(name);
         }
         starter.setUnsafeExperimentalButtonlessServiceInSecureDfuEnabled(true);
-        starter.setZip(filePath);
+
+        if (filePath.endsWith(".bin") || filePath.endsWith(".hex")) {
```

Comment:
On iOS the extension check is case insensitive, but on Android it isn't.  Wouldn't it be better to check against the filePath's toLowerCase()?